### PR TITLE
Add extern support in context json

### PR DESCRIPTION
**Pull request by usha1830**
This PR is to add target name details for externs as described in this CR
https://wiki.ith.intel.com/pages/viewpage.action?pageId=2253848776#ChangeRequests-Change_Request_CR00021
[context_all.json.txt](https://github.com/usha1830/p4c/files/9086320/context_all.json.txt)
[psa-example-dpdk-externs.p4.txt](https://github.com/usha1830/p4c/files/9086321/psa-example-dpdk-externs.p4.txt)

### CODE REVIEW

**Review by kamleshbhalui — APPROVED** [review]

File: backends/dpdk/dpdkContext.cpp [hunk 1]
```diff
@@ -109,6 +109,45 @@ void DpdkContextGenerator::CollectTablesAndSetAttributes() {
...
+                externTypeName == "Meter" ||
+                externTypeName == "Hash" ||
+                externTypeName == "InternetCheckSum") {
+                struct externAttributes externAttr;
+                externAttr.externalName = ed->controlPlaneName();
+                externAttr.externType = externTypeName;
+                if (externTypeName == "Counter") {
+                    if (ed->arguments->size() != 2) {
+                        ::error(ErrorType::ERR_EXPECTED,
+                                "%1%: expected number of counters and type of counter"
```
Comment (psivanup) [inline backends/dpdk/dpdkContext.cpp]: Isn't it unexpected number of arguments? There is no mention of number of arguments expected in the error message.

**Review by psivanup — COMMENTED** [review]
Please check the comment.

File: backends/dpdk/dpdkContext.cpp (hunk 1 above)
Comment (usha1830) [reply backends/dpdk/dpdkContext.cpp]: Tried to give little more detail about the expected argument

Comment (usha1830) [reply backends/dpdk/dpdkContext.cpp]: Oh you mean the error type?

Comment (usha1830) [reply backends/dpdk/dpdkContext.cpp]: Updated.

**Review by psivanup — APPROVED** [review]
Looks fine to me.

**Comment by usha1830** [comment]
Closing as upstream PR created

**Pull request closed without merging**